pycbc/inference/sampler/cpnest.py

```diff
@@ -0,0 +1,219 @@
+        obj = cls(model, nlive=nlive, maxmcmc=maxmcmc,
+                  nthreads=nthreads)
+        # set target
+        #obj.set_target_from_config(cp, section)
+        # add burn-in if it's specified
+        #obj.set_burn_in_from_config(cp)
+        return obj
+
+    def checkpoint(self):
+        pass
+
+    def finalize(self):
+        logz = self._sampler.NS.logZ
+        dlogz = 0.1 #######FIXME!!!!!###############
+        logging.info("log Z, dlog Z: {}, {}".format(logz, dlogz))
+        for fn in [self.checkpoint_file]:
+            with self.io(fn, "a") as fp:
+                fp.write_logevidence(logz,dlogz)
+        self.write_results(self.checkpoint_file)
+        logging.info("Writing samples to files")
+	for fn in [self.checkpoint_file, self.backup_file]:
+	    self.write_results(fn)
+	    with self.io(fn, "a") as fp:
+	        fp.write_niterations(self.niterations)
+	logging.info("Validating checkpoint and backup files")
+	checkpoint_valid = validate_checkpoint_files(
+	    self.checkpoint_file, self.backup_file)
+	if not checkpoint_valid:
+	    raise IOError("error writing to checkpoint file")
+
+    @property
+    def model_stats(self):
+        logl = self._sampler.posterior_samples['logL']
+        # get prior from posterior
+        logp = self._sampler.posterior_samples['logPrior']
+        return {'loglikelihood': logl, 'logprior': logp}
+
+    @property
+    def samples(self):
+        samples_dict = {p: self._sampler.posterior_samples['%s'%p] for p in
+	                        self.posterior_samples.dtype.names}
+	return samples_dict
+
+    def set_initial_conditions(self, initial_distribution=None,
+                               samples_file=None):
+        """Sets up the starting point for the sampler.
+
+        Should also set the sampler's random state.
+        """
+        pass
+    
+    def write_results(self, filename):
+        """Writes samples, model stats, acceptance fraction, and random state
+        to the given file.
+        Parameters
+        -----------
+        filename : str
+            The file to write to. The file is opened using the ``io`` class
+            in an an append state.
+        """
+        with self.io(filename, 'a') as fp:
+            # write samples
+            print 'filename is', filename
+            fp.write_samples(self.samples, self.model.variable_params)
+            # write stats
+            fp.write_samples(self.model_stats)
+            # write accpetance
+            #fp.write_log_evidence(self._sampler.NS.logZ)
+            # write random state
+            #fp.write_random_state()
+
+    @property
+    def nested_samples(self):
+        return self._sampler.nested_samples
+    
+    @property
+    def posterior_samples(self):
+        return self._sampler.posterior_samples
+
+    @property
+    def logz(self):
+        return self._logz
+
+    @property
+    def dlogz(self):
+        return self._dlogz
+
+class CPnestModel(cpm.Model):
```

Comment:
Add a description of this. It can just be a quick one liner since this is only used internally.